src/EditorFeatures/CSharpTest/ConvertToRecord/ConvertToRecordTests.cs

```diff
@@ -0,0 +1,1690 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertToRecord;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ConvertToRecord
+{
+    using VerifyCS = CSharpCodeRefactoringVerifier<CSharpConvertToRecordRefactoringProvider>;
+
+    [Trait(Traits.Feature, Traits.Features.CodeActionsConvertToRecord)]
+    public class ConvertToRecordTests
+    {
+        [Fact]
+        public async Task TestNoProperties_NoAction()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class [|C|]
+    {
+        // field, not property
+        public int f = 0;
+    }
+}
+";
+            await TestNoRefactoringAsync(initialMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestExplicitProperty_NoAction1()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class [|C|]
+    {
+        private int f;
+
+        public int P 
+        {
+            get => f; 
+            init => f = value;
+        }
+    }
+}
+";
+            await TestNoRefactoringAsync(initialMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestExplicitProperty_NoAction2()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class [|C|]
+    {
+        private int f;
+
+        public int P 
+        {
+            get
+            {
+                return f;
+            }
+
+            init
+            {
+                f = value;
+            }
+        }
+    }
+}
+";
+            await TestNoRefactoringAsync(initialMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestExplicitProperty_NoAction3()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class [|C|]
+    {
+        public int P { get; init; } = 4;
+    }
+}
+";
+            await TestNoRefactoringAsync(initialMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestReadonlyProperty_NoAction()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class [|C|]
+    {
+        public int P { get; }
+    }
+}
+";
+            await TestNoRefactoringAsync(initialMarkup).ConfigureAwait(false);
```

Comment:
this seems odd.  what woudl be wrong with offering a refactoring here?  seems like a nice smiple data class that someone might say "yeah, i'd prefer this as a record now htat c# has them".

Reply:
I agree. I didn't realize that not including a set still allowed the class to initialize in the constructor, although I'm not sure what I was thinking it was for at all if that wasn't possible.